src/extension.ts

```diff
@@ -96,18 +166,37 @@ class PyPICodeLens extends vscode.CodeLens {
 
 class PyPICodeLensProvider implements vscode.CodeLensProvider<PyPICodeLens> {
     provideCodeLenses(document: vscode.TextDocument): PyPICodeLens[] {
+        const codeLensEnabled = vscode.workspace.getConfiguration('pypiAssistant').get('codeLens')
         const codeLenses: PyPICodeLens[] = []
-        if (vscode.workspace.getConfiguration('pypiAssistant').get('codeLens')) {
+
+        if (!codeLensEnabled) {
+            return codeLenses
+        }
+
+        if (document.languageId === 'toml' && document.fileName.toLowerCase().endsWith('pyproject.toml')) {
+            const dependencies = getPyProjectDependencies(document.getText())
+
             for (let line = 0; line < document.lineCount; line++) {
-                let requirement: Requirement | null
-                try {
-                    requirement = parsePipRequirementsLineLoosely(document.lineAt(line).text)
-                } catch {
-                    continue
-                }
+                const lineText = document.lineAt(line).text
+                const requirement = dependencies.find((dependency) => {
+                    return lineText.startsWith(`${dependency.name} = `)
+                })
+
                 if (requirement?.type !== 'ProjectName') continue
                 codeLenses.push(new PyPICodeLens(new vscode.Range(line, 0, line, 0), requirement))
             }
+            return codeLenses
+        }
+
+        for (let line = 0; line < document.lineCount; line++) {
```

Comment:
Same as in the hover provider, the pyproject.toml and pip-requirements branches would be more maintainable as separate methods